### ISSUE TITLE
Either We Need a Tool That Corrects the Misaligned Bone Angles of Certain Ragdolls, or at the Bare Minimum, Strongly Encourage Addon Makers to Release the QC Source Codes

### ISSUE THREAD

**Issue by ToxicHolyGrenade** [Facepunch/garrysmod-issues#5739]
### Details

https://github.com/Winded/RagdollMover/issues/27

https://github.com/Winded/RagdollMover/issues/27#issuecomment-1910219192

In accordance to these links, there are certain ragdolls that, through the Ragdoll Mover tool by Winded, have misaligned bone angles, like the TF2 Scout ragdolls.

In consequence, this puts me and other Gmod users into a difficult position where we have to resort to outdated measures like the physgun for machinima making, which can be a really sad state of affairs, which is the last thing we want.

We need a tool that through copied QC bone values relating to other ragdolls, we can get rid of this problem once and for all.

If that cannot be done, then it must be essential to strongly encourage addon creators to release source codes more often, especially QC files relating to ragdolls.

### Steps to reproduce
Given that certain ragdolls have misaligned bone angles, and aren't straight enough, it shouldn't be too hard to emulate this problem via the Ragdoll Mover.

**Comment by robotboy655** [comment]
I am really not sure what you are asking here? It sounds like a problem with either the models, or the tool. What do you want from the base game? It was nothing to do with either.

**Comment by ToxicHolyGrenade** [comment]
That's more so from the models, but how should we fix these issues? Because this is going on for far too long. To be honest, this has nothing to do with the Ragdoll Mover as I have found out, it is the problem with the bone angles being disorientated, like the pelvises.

**Comment by robotboy655** [comment]
You don't I guess. If the problem is with the models, the models themselves need to be fixed.

**Comment by ToxicHolyGrenade** [comment]
> You don't I guess. If the problem is with the models, the models themselves need to be fixed.

Then should we strongly encourage addon creators to always release their source codes in case if something needs to be fixed? Especially the ragdolls? Besides, is there even a way to splice the QC codings from both ragdolls and models?

**Comment by garryspins** [comment]
strongly encourage how

**Comment by MrTheshadowon** [comment]
And too be fair, I'm not sure he knows what a .qc file is, nor am I certain that he has ever decompiled a model from the source engine before.

**Comment by ToxicHolyGrenade** [comment]
> ![image](https://private-user-images.githubusercontent.com/156720891/307409701-aba6756a-28c6-4fc6-91d5-e8e513af1547.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg5NzQzNzEsIm5iZiI6MTcwODk3NDA3MSwicGF0aCI6Ii8xNTY3MjA4OTEvMzA3NDA5NzAxLWFiYTY3NTZhLTI4YzYtNGZjNi05MWQ1LWU4ZTUxM2FmMTU0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNlQxOTAxMTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZGJmZGY2NWM3MWRmNGIwYjVjOTQ0ODU1NmIyNzdjZmFjYjE4Zjk0YmU0ODVmN2FmZTExODAzMmI0MDM3MGNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Hk0G-m0susjixymMUgxdcr8Uh8tlBl3cW26gjfQJdxo) As you can see here, we have two options for ragdoll mover, the one he had checkmarked was "Localized Position Gizmo" which in translation allows the user to use the world axis instead of the model's axis. ![image](https://private-user-images.githubusercontent.com/156720891/307410710-844e4086-8efe-430b-8580-7dc1f9df75ab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg5NzQzNzEsIm5iZiI6MTcwODk3NDA3MSwicGF0aCI6Ii8xNTY3MjA4OTEvMzA3NDEwNzEwLTg0NGU0MDg2LThlZmUtNDMwYi04NTgwLTdkYzFmOWRmNzVhYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNlQxOTAxMTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZGZlZDFjMWJjMTc5NmY4ZTFlNmVhOTNiMzJiNjY1ZWQ2NGEyZWU0MGI3ZWVjZTcxZDgzOWI2NDIzNjhkYTRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.gVnGm9Zm9v0m9s52z_fyL2__BYtdHbstpQZ3eB6UYII)
> 
> He was complaining that the axis is misaligned when it really isn't, he was just on the world axis, and then he goes off on a tangent on how we should incorporate .qc files to ragdoll mover.
> 
> It isn't going to work, there is no issue with the addon, the addon is fine.



> There is literally no problem with the addon, I don't know why he's constantly saying there is a problem. I feel like this is just trying to break the addon on purpose.

You don't understand, I know there is nothing wrong with the addon, but what I am saying, is that there is something fundamentally off regarding the pelvis bones of various Scout ragdolls whenever it comes to the .qc files used to create them.

And what you are saying is you jumping on the bandwagon of shooting my reported issues down without seeing the bigger picture. So long as you have that type of mentality, that will never bring anything productive to the community. It it important to hear someone out, consider what they are saying, before making a judgement.

Long story short, I am not talking about the Ragdoll Mover tool, I am talking about the bone aspects of the ragdolls when it comes to .qc files, which results in the problem I am showing.
